File: src/main/java/leetcode/TwoSum.java
package leetcode;

import java.util.HashMap;
import java.util.Map;

public class TwoSum {

    public static Integer[] twoSum(Integer[] items, Integer target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < items.length; i++) {
            Integer x = target - items[i];
            if (map.containsKey(x))
                return new Integer[]{map.get(x), i};
            map.put(items[i], i);
        }
        return null;
    }

}

File: src/main/java/leetcode/TwoSum.kt
package leetcode

fun twoSum(items: Array<Int>, target: Int): Array<Int?> {
    val map = HashMap<Int, Int>()
    /*for ((index, element) in items.withIndex()) {
        val x = target - index
        if (map.containsKey(x))
            return arrayOf(map[x], index)
        map[element] = index
    }*/
    items.forEachIndexed { index, element ->
        val x = target - index
        if (map.containsKey(x))
            return arrayOf(map[x], index)
        map[element] = index
    }
    return emptyArray();
}